auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -184,6 +184,20 @@ public static AuthUI getInstance() {
      */
     @NonNull
     public static AuthUI getInstance(@NonNull FirebaseApp app) {
+        AuthUI authUi;
+        synchronized (INSTANCES) {
+            authUi = INSTANCES.get(app);
+            if (authUi == null) {
+                authUi = new AuthUI(app);
+                authUi.mAuth.setFirebaseUIVersion(BuildConfig.VERSION_NAME);
```

Comment:
Is there anything else to this method? I've noticed that it returns a task (should we be listening for something?) and takes in a nullable parameter (is there a set/unset lifecycle?).

PS: Looks like there are conflicting annotations 😕:
![image](https://user-images.githubusercontent.com/9490724/34284122-9a33c914-e684-11e7-8025-cd5f2c886b88.png)
![image](https://user-images.githubusercontent.com/9490724/34284131-a73ad58a-e684-11e7-947a-260df87b07a6.png)

Reply:
I think we can use it as fire-and-forget.